src/ol_infrastructure/components/services/k8s.py

```diff
@@ -0,0 +1,311 @@
+                    f"./manage.py promote_user --promote --superuser '{mit_username}@mit.edu'; exit 0",  # noqa: E501
+                ],
+                image_pull_policy="IfNotPresent",
+                env=application_deployment_env_vars,
+                env_from=application_deployment_envfrom,
+            )
+            for mit_username in DEVOPS_MIT
+        ]
+
+        # Create a deployment resource to manage the application pods
+        application_labels = self.ol_app_k8s_config.k8s_global_labels | {
+            "ol.mit.edu/application": "unified-application",
+            "ol.mit.edu/pod-security-group": "application-app",
+        }
+
+        _application_deployment = kubernetes.apps.v1.Deployment(
+            f"unified-application-{stack_info.env_suffix}-deployment",
+            metadata=kubernetes.meta.v1.ObjectMetaArgs(
+                name="application-app",
+                namespace=self.ol_app_k8s_config.application_namespace,
+                labels=application_labels,
+            ),
+            spec=kubernetes.apps.v1.DeploymentSpecArgs(
+                # TODO @Ardiea: Add horizontial pod autoscaler  # noqa: TD003, FIX002
+                replicas=1,
+                selector=kubernetes.meta.v1.LabelSelectorArgs(
+                    match_labels=application_labels,
+                ),
+                # Limits the chances of simulatious pod restarts -> db migrations
+                # (hopefully)
+                strategy=kubernetes.apps.v1.DeploymentStrategyArgs(
+                    type="RollingUpdate",
+                    rolling_update=kubernetes.apps.v1.RollingUpdateDeploymentArgs(
+                        max_surge=0,
+                        max_unavailable=1,
+                    ),
+                ),
+                template=kubernetes.core.v1.PodTemplateSpecArgs(
+                    metadata=kubernetes.meta.v1.ObjectMetaArgs(
+                        labels=application_labels,
+                    ),
+                    spec=kubernetes.core.v1.PodSpecArgs(
+                        volumes=[
+                            kubernetes.core.v1.VolumeArgs(
+                                name="staticfiles",
+                                empty_dir=kubernetes.core.v1.EmptyDirVolumeSourceArgs(),
+                            ),
+                            kubernetes.core.v1.VolumeArgs(
+                                name="nginx-config",
+                                config_map=kubernetes.core.v1.ConfigMapVolumeSourceArgs(
+                                    name=application_nginx_configmap.metadata.name,
+                                    items=[
+                                        kubernetes.core.v1.KeyToPathArgs(
+                                            key="web.conf",
+                                            path="web.conf",
+                                        ),
+                                    ],
+                                ),
+                            ),
```

Comment:
We should probably revisit the use and utility of nginx in our apps as they exist in K8s